provider/ec2/local_test.go

```diff
@@ -2410,18 +2413,26 @@ func (t *localServerSuite) TestStopInstances(c *gc.C) {
 	// for Instances to return an error, and it will not retry
 	// if it succeeds.
 	gone := false
-	for a := ec2.ShortAttempt.Start(); a.Next(); {
+	retryStrategy := retry.CallArgs{
+		Clock:       clock.WallClock,
+		MaxDuration: 5 * time.Second,
+		Delay:       200 * time.Millisecond,
+	}
+	retryStrategy.Func = func() error {
 		insts, err = t.Env.Instances(t.callCtx, []instance.Id{inst0.Id(), inst2.Id()})
 		if err == environs.ErrPartialInstances {
 			// instances not gone yet.
-			continue
+			return err
```

Comment:
Don't we need to indicate that this error is not fatal? I.e. keep going until `ErrNoInstances` or time-out.

Reply:
Errors are not fatal by default inside of retries. So this will do exactly that

Reply:
Although this does highlight I can probably use errs to drop the weird gone var here